src/core/server/logging/README.md

```diff
@@ -127,6 +131,138 @@ Outputs the process ID.
 With `json` layout log messages will be formatted as JSON strings that include timestamp, log level, context, message 
 text and any other metadata that may be associated with the log message itself.
 
+## Appenders
+
+### Rolling File Appender
+
+Similar to Log4j's `RollingFileAppender`, this appender will log into a file, and rotate it following a rolling
+strategy when the configured policy triggers.
+
+#### Triggering Policies
+
+The triggering policy determines when a rolling should occur.
+
+There are currently two policies supported: `size-limit` and `time-interval`.
+
+##### SizeLimitTriggeringPolicy
+
+This policy will rotate the file when it reaches a predetermined size.
+
+```yaml
+logging:
+  appenders:
+    rolling-file:
+      kind: rolling-file
+      path: /var/logs/kibana.log
+      policy:
+        kind: size-limit
```

Comment:
Well, IDK about doing it directly in this PR instead of globalizing all the required changes in https://github.com/elastic/kibana/issues/57551 tbh. We would be introducing inconsistencies in our own logging API, which is imho probably worse.

In example:

> size-limit --> SizeBasedTriggeringPolicy

All our other 'types' are lowercase, dash separated. Having a mix of both seems wrong

> policy --> policies

In log4j configuration , `policies` is a map (they accept multiple policies). we don't ATM (and TIL by looking at log4j's code, it would requires significant work, their code is a patchwork of hacks and side effects to support that...) I feel like using `policies` could be misleading as the user may try to use an array instead of an object in the configuration?

Also, note that if we want to align the naming, there are other things to change:

`kind: size-limit`, in log4j, it's `type`, not `kind`. However all our other logging configuration structures are using the `kind` naming, so this would introduce yet more divergence in our own logging config API.

Not opposed to do it right now if you think it's still better though, just tell me.

Reply:
>  in log4j, it's type, not kind

yeah, https://github.com/elastic/kibana/issues/57551 should fix it. 

> All our other 'types' are lowercase, dash separated.

Just curious, what `types` do you mean? I'm fine if we postpone this change till https://github.com/elastic/kibana/issues/57551 also it means that we should do that before removing the legacy log rotation @joshdover any objections?